lib/iris/tests/unit/common/lenient/test_Lenient.py

```diff
@@ -4,179 +4,166 @@
 # See LICENSE in the root of the repository for full licensing details.
 """Unit tests for the :class:`iris.common.lenient.Lenient`."""
 
-# Import iris.tests first so that some things can be initialised before
-# importing anything else.
-import iris.tests as tests  # isort:skip
+import pytest
 
-from unittest.mock import sentinel
+from iris.common.lenient import _LENIENT, _LENIENT_PROTECTED, Lenient
 
-from iris.common.lenient import _LENIENT, Lenient
 
+@pytest.fixture()
+def lenient():
+    # setup
+    state = {key: _LENIENT.__dict__[key] for key in _LENIENT_PROTECTED}
+    # call
+    yield Lenient()
+    # teardown
+    for key, value in state.items():
+        _LENIENT.__dict__[key] = value
 
-class Test___init__(tests.IrisTest):
-    def test_default(self):
-        lenient = Lenient()
+
+class Test___init__:
+    def test_default(self, lenient):
         expected = dict(maths=True)
-        self.assertEqual(expected, lenient.__dict__)
+        assert expected == lenient.__dict__
```

Comment:
I think ideally this should go the other way around.
Ideally, all the pytest style equality checks should by "assert actual == expected" and not "expected == actual".
I believe this matters only because if it fails, in some cases the message could be a bit incomprehensible if the order is wrong.

N.B. the existing "way around" **_is_** correct for unittest , as in "assertEqual(expected, ...)".  Personally I always found that a bit non-obvious anyway.

Please treat as a general point : I'm not going to stick comments everywhere just for this !